cli/tests/test_metaedit_command.rs

```diff
@@ -485,6 +485,38 @@ fn test_update_empty_author() {
     ");
 }
 
+#[test]
+/// Test that setting the same timestamp twice does nothing (issue #7602)
+fn test_metaedit_set_same_timestamp_twice() {
+    let test_env = TestEnvironment::default();
+    test_env.run_jj_in(".", ["git", "init", "repo"]).success();
+    let work_dir = test_env.work_dir("repo");
+
+    // Set the author-timestamp to the same value twice
+    // and check that the second time it does nothing
+    work_dir
+        .run_jj([
+            "metaedit",
+            "--author-timestamp",
+            "2001-02-03 04:05:14.000+07:00",
+        ])
+        .success();
```

Comment:
nit: snapshot the output as well (instead of `.success())`? I think it's better to ensure that the commit is rewritten by this command.